fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/MessageValidator.kt

```diff
@@ -231,6 +254,28 @@ object MessageValidator {
       return this
     }
 
+    /**
+     * Helper method to check that a message has a valid structure.
+     *
+     * @param message the message to check, expected to be a map with string keys and optional
+     *   string values
+     */
+    fun validMessage(message: Any): MessageValidatorBuilder {
+      message as Map<*, *>
+
+      val data = message["data"] as String?
+      val sender = message["sender"] as String?
+      val signature = message["signature"] as String?
+      val messageId = message["message_id"] as String?
```

Comment:
If you manage to cast that `Any `to `MessageGeneral `(or create another class for it), then you'd already have the fields in the message object without needing to access it as a map.